lib/services/volunteer.js

```diff
@@ -2,137 +2,80 @@ const Schmervice = require('schmervice');
-    let months;
-    months = (d2.getFullYear() - d1.getFullYear()) * 12;
-    months -= d1.getMonth();
-    months += d2.getMonth();
-    if (months <= 0) {
-      return 0;
-    }
-    return months;
-  }
-
   async getAllVolunteers() {
-    const { UserRole, User, Classes } = this.server.models();
-    const volunteers_data = {};
+    const { UserRole, Classes } = this.server.models();
+    let volunteers = [];
 
     try {
-      // get user_ids from userRole
-      const volunteer_ids = await UserRole.query()
+      // get volunteer data and partner data
+      volunteers = await UserRole.query()
         .select('user_id as id')
-        .where('role', 'volunteer');
+        .where('role', 'volunteer')
+        .withGraphFetched('user.[partner]');
 
-      /**
-       * You should use withGraphFetched function of Objection to leverage the relation mapping of UserRole model
-       * with Users model further with classes model
-       *
-       * Something like this
-       * const volunteers = await UserRole.query().where('role', 'volunteer').withGraphFetched('user');
-       *
-       * Just take care of one thing Classes -> User relation mapping is done on keys classes.user_id -> users.id, so this will give you the data
-       * of students joining the class and not the facilitators
-       *
-       */
-
-      volunteer_ids.forEach((v, index) => {
-        volunteer_ids[index] = v.id;
+      volunteers.forEach((v, index) => {
+        volunteers[index] = volunteers[index].user;
       });
 
-      // query Users for name
-      const volunteers = await User.query().select('name', 'id').whereIn('id', volunteer_ids);
-
-      // get classes details
+      // get classes and ratings data
       const classes_data = await Promise.all(
         volunteers.map((volunteer) => {
           return Classes.query()
             .where('facilitator_id', volunteer.id)
-            .orderBy('start_time', 'desc');
+            .orWhere('facilitator_email', volunteer.email)
+            .withGraphFetched('[pathways.pathway, feedbacks]');
         })
       );
 
-      volunteers.forEach((volunteer, i) => {
-        const classes = classes_data[i];
-        if (classes.length > 0) {
-          const last_class = classes[0];
-          const first_class = classes[classes.length - 1];
-          volunteer.last_class_date = last_class.start_time;
-          volunteer.last_class_title = last_class.title;
-          volunteer.engagement_in_months = UserRoleService.calculateEngagement(
-            first_class.start_time,
-            last_class.start_time
-          );
-          volunteer.lang = last_class.lang;
-        } else {
-          volunteer.last_class_date = null;
-          volunteer.last_class_title = null;
-          volunteer.engagement_in_months = 0;
-          volunteer.lang = null;
-          volunteer.engagement_in_months = 0;
-        }
+      volunteers.forEach((v, i) => {
+        volunteers[i].classes = classes_data[i];
```

Comment:
```suggestion
        v.classes = classes_data[i];
```
Would need `volunteers[i]` if you were changing the items in the Array themselves, but if you're only changing the `classes` property of each item, you can use `v` (Illustrated below):

`v -> { classes: ___, ... } <- volunteers[i]`

Changing `v` to refer to a different object would have no effect on `volunteers[i]` as it would continue to point to the same object, but changing the `classes` property of the object `v` refers to updates this property for `volunteers[i]` as well since they both refer to the same object.